server/server_test.go

```diff
@@ -0,0 +1,120 @@
+package server_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+
+	pbExample "github.com/gogo/grpc-example/proto"
+	"github.com/gogo/grpc-example/server"
+)
+
+//go:generate mockgen -destination mocks_test.go -package server_test github.com/gogo/grpc-example/proto UserService_ListUsersServer,UserService_ListUsersByRoleServer
```

Comment:
First time seeing golang/mock in action.  The reflection (in the generated code) and the timing (setting up the expected before doing the call), is a bit weird, but I see that it saves a bit of boiler plate code and if this is the standard way of mocking in golang, then its good to showcase this as well.
Thanks for teaching me something :)

Reply:
At work we actually use [ginkgo](https://onsi.github.io/ginkgo/) and [gomega](https://onsi.github.io/gomega/) coupled with either gomock or [moq](github.com/matryer/moq). For these tests I wanted to stay reasonably close to the stdlib and I saw https://github.com/grpc/grpc-go/blob/master/Documentation/gomock-example.md which seems to favor this structure.